lesson14/sahra/script.js

```diff
@@ -0,0 +1,141 @@
+
+// 4. Declare a variable named `firstName` and initialize it to your first name as a string.
+// Declare a variable named `lastName` and initialize it to your last name as a string.
+// Declare a variable named `fullName` and initialize it to the concatenation of `firstName` and `lastName`.
+
+console.log('Exercise 4:');
+
+const firstName = 'Ayse';
+const lastName = 'Baskin';
+const fullName = firstName + ' ' + lastName;
+console.log(fullName);
+
+console.log('------------------------');
+
+// 5. Declare a variable named `firstFactor` and initialize it to 7.
+// Declare a variable named `secondFactor` and initialize it to 2.
+// Declare a variable named `multiplicationResult` and initialize it to the product of `firstFactor` and `secondFactor`.
+
+const firstFactor = 7;
+const secondFactor = 2;
+const multiplicationResult = firstFactor * secondFactor;
+
+console.log('Exercise 5:');
+console.log(multiplicationResult);
+
+// 6. Declare a variable named `PI` and initialize it to the value of PI (3.14159). Use the built-in function `Math.PI`.
+
+console.log('Exercise 6:', Math.PI);
+console.log('------------------------');
+
+// 7. Declare a variable named `counter` and initialize it to 0.
+// Increment the `counter` variable by 1 using 3 different ways to increment numbers.
+let counter = 0;
+const step = 1;
+
+console.log((counter = counter + step));
+
+console.log(counter++);
+
+console.log((counter += step)); // counter = counter + step;
+
+console.log(++counter);
+console.log('------------------------');
+
+// 8. Declare a variable named `initialTemperature` and initialize it to 20.
+// Increase the value of `initialTemperature` by 5 and log the result to the console.
+// Global scope
+let initialTemperature = 20;
+const temperatureStep = 5;
+
+console.log((initialTemperature = initialTemperature + temperatureStep));
+
+console.log('Exercise 8:');
+console.log('------------------------');
+
+// 9. Declare a variable named `numberEx9` and initialize it to 6.
+// Declare a variable named `incrementedNumber` and initialize it to `++numberEx9` (numberEx9 incremented using the prefix ++ operator).
+// Log both values to the console.
+let numberEx9 = 6;
+const incrementedNumber = ++numberEx9;
+
+console.log(numberEx9, incrementedNumber);
+console.log('------------------------');
+
+// 10. Declare a variable named `numberEx10` and initialize it to 8.
+// Declare a variable named `postIncrementedNumber` and initialize it to `numberEx10++` (numberEx10 incremented using the postfix ++ operator).
+// Log both values to the console.
+let numberEx10 = 8;
+const postIncrementedNumber = numberEx10++;
+
+console.log(numberEx10, postIncrementedNumber);
+console.log('------------------------');
+
+// 11. Declare a variable named `numberEx11` and initialize it to -3.
+// Use the prefix ++ operator to increment the value of `numberEx11` by 1, then multiply the result by 2 and log it to the console.
+let numberEx11 = -3;
+var result = (numberEx11 + 1) * 2;
```

Comment:
You try to use variable name result in both ex. 11 and 12. Since the variable is defined in the global scope, you are actually re-using the same variable instead of creating a new one. If you used 'const' keyword, it would warn you about this issue.

Rename 'result' here to 'resultEx11' (or similar), and use 'const' instead of 'let'.

Reply:
Also, you need to use '++' operator here to increment 'numberEx11'. You can solve this on multiple lines:
```
numberEx11++;
const resultEx11 = numberEx11 * 2;
```